src/index.ts

```diff
@@ -1,31 +1,144 @@
 const REGION_IDS = ["auk", "bop", "can", "cit", "gis", "hkb", "mwt", "mbh", "nsn", "ntl", "ota", "stl", "tki", "tas", "wko", "wgn", "wtc"]
 
 interface DistrictData {
-  color: string
+  color: string,
   description?: string
 }
 
 interface MapOptions {
   width?: number,
   height?: number,
+  targetElement: string,
   data?: {
     [key in typeof REGION_IDS[number]]: DistrictData
   }
 }
 
 class NZMapSVG {
-  options: MapOptions
+  options: MapOptions;
+
+  // Element tags
+  wrapper: any; //<div>
+  mapImage: any; //<svg>
 
   constructor(options: MapOptions) {
-    this.options = options
+    this.options = options;
   }
 
   // Add other methods here
+  render() {
+    if (
+      !this.options.targetElement ||
+      !document.getElementById(this.options.targetElement)
+    ) {
+      this.error('Target element not found');
+    }
+
+    // <svg> inside <div id='map'>
+    this.wrapper = document.getElementById(this.options.targetElement);
+    this.mapImage = document.createElementNS(
+      'http://www.w3.org/2000/svg',
+      'svg'
+    );
+    this.wrapper.appendChild(this.mapImage);
+
+    this.mapImage.setAttribute('viewBox', '0 0 701.42358 988.42462');
+    this.mapImage.setAttribute('fill', 'white');
+    this.mapImage.setAttribute('stroke', 'black');
+    this.mapImage.classList.add('svgMap-map-image');
+
+
+    // <path> inside <svg>
+    Object.keys(this.mapPaths).forEach(
+      function (countryID) {
+        var countryData = this.mapPaths[countryID];
+        if (!countryData.d) {
+          return;
+        }
+
+        var countryElement = document.createElementNS(
```

Comment:
regionElement

Reply:
yup this was already changed in my latest version that I haven't pushed up yet